homeassistant/components/azure_devops/sensor.py

```diff
@@ -2,89 +2,227 @@
 
 from __future__ import annotations
 
-from collections.abc import Callable
+from collections.abc import Callable, Mapping
 from dataclasses import dataclass
+from datetime import datetime
+import logging
 from typing import Any
 
 from aioazuredevops.builds import DevOpsBuild
 
-from homeassistant.components.sensor import SensorEntity, SensorEntityDescription
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+)
 from homeassistant.config_entries import ConfigEntry
 from homeassistant.core import HomeAssistant
 from homeassistant.helpers.entity_platform import AddEntitiesCallback
 from homeassistant.helpers.typing import StateType
+from homeassistant.util import dt as dt_util
 
 from . import AzureDevOpsDeviceEntity, AzureDevOpsEntityDescription
 from .const import CONF_ORG, DOMAIN
 
+_LOGGER = logging.getLogger(__name__)
+
 
 @dataclass(frozen=True, kw_only=True)
-class AzureDevOpsSensorEntityDescription(
-    AzureDevOpsEntityDescription, SensorEntityDescription
-):
-    """Class describing Azure DevOps sensor entities."""
+class AzureDevOpsBaseBuildSensorEntityDescription(SensorEntityDescription):
+    """Class describing Azure DevOps base build sensor entities."""
+
+    attrs: Callable[[DevOpsBuild], dict[str, Any]] | None
+    value: Callable[[DevOpsBuild], datetime | StateType]
 
-    build_key: int
-    attrs: Callable[[DevOpsBuild], Any]
-    value: Callable[[DevOpsBuild], StateType]
+
+@dataclass(frozen=True, kw_only=True)
+class AzureDevOpsBuildSensorEntityDescription(
+    AzureDevOpsEntityDescription,
+    AzureDevOpsBaseBuildSensorEntityDescription,
+):
+    """Class describing Azure DevOps build sensor entities."""
+
+    item_key: int = 0
```

Comment:
These 2 can be merged now, as in, there is no technical reason anymore to split them